src/compiler.ts

```diff
@@ -3460,6 +3468,18 @@ export class Compiler extends DiagnosticEmitter {
         }
         fromType = fromType.nonNullableType;
       }
+      // When we convert from the closure type into a function pointer, we first
+      // update the local copy of the scope with the newest values
+      if(fromType.is(TypeFlags.CLOSURE)) {
+        //Retain the Closure Class memory so it isn't freed when we change it into a function ptr
+        if (!this.skippedAutoreleases.has(expr)) {
+          let index = this.tryUndoAutorelease(expr, this.currentFlow);
+          if (index == -1) expr = this.makeRetain(expr);
+          this.skippedAutoreleases.add(expr);
+        }
```

Comment:
I'm not sure how we'll eventually free the closure given that we lose track of its size- that's another major missing piece